### PR TITLE
ConsumeMmapPerfEvent now correctly initializes filename

### PR DESCRIPTION
**Pull request by dimitry-**
Also rename modules_update_event field to module_update_event in
CaptureEvent proto

Test: Build, run local capture on OrbitUi

### CODE REVIEW

File: OrbitService/LinuxTracingHandler.cpp [hunk 1]
```diff
@@ -144,9 +144,9 @@ void LinuxTracingHandler::OnTracepointEvent(orbit_grpc_protos::TracepointEvent t
 }
 
 void LinuxTracingHandler::OnModulesUpdate(
-    orbit_grpc_protos::ModuleUpdateEvent modules_update_event) {
+    orbit_grpc_protos::ModuleUpdateEvent module_update_event) {
```
Comment (dpallotti) [inline OrbitService/LinuxTracingHandler.cpp:147]: Can you rename it also in the header file and in `TracerListener`?

File: OrbitLinuxTracing/PerfEventReaders.cpp [hunk 2]
```diff
@@ -81,7 +81,9 @@ std::unique_ptr<MmapPerfEvent> ConsumeMmapPerfEvent(PerfEventRingBuffer* ring_bu
       header.size - filename_offset - sizeof(perf_event_sample_id_tid_time_streamid_cpu);
   std::vector<char> filename_vector(filename_size);
   ring_buffer->ReadRawAtOffset(&filename_vector[0], filename_offset, filename_size);
-  std::string filename(&filename_vector[0], filename_size);
+  // This is a bit paranoid but you never know
+  filename_vector[filename_size - 1] = '\0';
+  std::string filename(&filename_vector[0]);
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventReaders.cpp:86]: I personally prefer `filename_vector.data()`.

File: OrbitLinuxTracing/PerfEventReaders.cpp [hunk 3]
```diff
@@ -81,7 +81,9 @@ std::unique_ptr<MmapPerfEvent> ConsumeMmapPerfEvent(PerfEventRingBuffer* ring_bu
       header.size - filename_offset - sizeof(perf_event_sample_id_tid_time_streamid_cpu);
   std::vector<char> filename_vector(filename_size);
   ring_buffer->ReadRawAtOffset(&filename_vector[0], filename_offset, filename_size);
-  std::string filename(&filename_vector[0], filename_size);
+  // This is a bit paranoid but you never know
+  filename_vector[filename_size - 1] = '\0';
```
Comment (dpallotti) [inline OrbitLinuxTracing/PerfEventReaders.cpp:85]: Is this the line that solves the problem with the logs? If yes, the comment should be different. Or what exactly was the issue?

Comment (dimitry-) [reply OrbitLinuxTracing/PerfEventReaders.cpp:85]: the next line solved the problem, this one is just me not trusting data provided by the kernel :)

**Review by florian-kuebler — APPROVED** [review]
I agree with Davide's comments. Despite that, LGTM!

File: OrbitLinuxTracing/PerfEventReaders.cpp (hunk 2 above)
Comment (dimitry-) [reply OrbitLinuxTracing/PerfEventReaders.cpp:86]: done.

File: OrbitService/LinuxTracingHandler.cpp (hunk 1 above)
Comment (dimitry-) [reply OrbitService/LinuxTracingHandler.cpp:147]: done.

**Review by dpallotti — APPROVED** [review]

**Pull request merged**